### DISCUSSION TITLE
[FEAT] : Integration with OpenAI’s Assistants

### DISCUSSION THREAD

**Discussion by dominpm** [open-webui/open-webui#5988] [Ideas]
**What is the problem?**

Currently there is no direct way to use an OpenAI’s Assistant through the UI.

**Possible fix**

Add integration with OpenAI’s Assistants API.
It would be optimal to just import an Assitant’s ID into the UI (only admins) to be able to chat with it.

**Comment by PlebeiusGaragicus** [comment]
I have changed my mind on this in the last few months - I'd REALLY like to be able to use the Assistants API (in a pipeline, at least).  At the very least, an example pipeline would be helpful as I'm running into issues with running asynchronous code in my pipeline.

To be specific, LangGraph is using the Assistants API "standard" to expose their graphs.  I have a working OI implementation as well as an agentic graph, but I'm having trouble connecting the two.

If Open WebUI had the ability to connect to the Assistants API, then we'd be able to use OpenAI's or our own hosted LangGraph agents.

**Reply by dentroai** [reply]
wait, there's a standardization for Assistants API and Openai Assistants and Langgraph both use it?

**Reply by PlebeiusGaragicus** [reply]
Perhaps I shouldn't call it a "standard."  But refer to the LangGraph [API docs](https://langchain-ai.github.io/langgraph/cloud/reference/api/api_ref.html) and then compare with OpenAI's.

**Comment by harrywang** [comment]
any update for assistant api support？
